### PR TITLE
Throw validation error instead of res with error

### PR DESCRIPTION
**Pull request by michaltk**
so we can handle the errors as we would like..

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -85,10 +96,7 @@ function validationRoute<
       // TODO: figure out which reporter we wanna use
       const report = PathReporter.report(result);
       if (result.isLeft()) {
-        return res.status(422).json({
-          success: false,
-          errors: report
-        });
+        throw new IoTsValidationError(JSON.stringify(report));
```
Comment (tikotzky) [inline src/index.ts]: Why stringify it?

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     /* Basic Options */
-    "target": "ES5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
+    "target": "es2017" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
```
Comment (tikotzky) [inline tsconfig.json]: Why did this need to change?

Comment (michaltk) [reply tsconfig.json]: bec if doesn't work to do `err instanceof IoTsValidationError` if target `ES5`.
err was always instance of `Error`

Comment (tikotzky) [reply tsconfig.json]: Do you need to go all the way to es2017 it es2015 is enough?
You don’t want users of the lib to need to transpile.

File: src/index.ts (hunk 1 above)
Comment (michaltk) [reply src/index.ts]: Message had to be a string..
but you mean I should rather stringify in the class?

Comment (tikotzky) [reply src/index.ts]: So perhaps the report can be added as a property on the error.
The message should prob be a more generic string. Perhaps a formatted version of the report. (I think iots has some built in report formatters)

File: tsconfig.json (hunk 2 above)
Comment (michaltk) [reply tsconfig.json]: Yes, that works too.. I'll update.

File: src/index.ts (hunk 1 above)
Comment (eligolding) [reply src/index.ts]: Their built in reporters are not that good.

Comment (eligolding) [reply src/index.ts]: Agreed the report should be a property.

**Pull request merged**